### PR TITLE
Upload avatar guided tour

### PR DESCRIPTION
**Pull request by markryall**
This PR adds a guided tour triggered from the onboarding checklist.

# Testing

* Visit `/checklist/<domain>` for your site
* Click 'Do It' for the user avatar task (whether completed or not - the tour should show)
* Verify that the following steps appear in the tour:

user-avatar

# Issues

* the tour does not trigger the first time the checklist item is clicked
* clicking 'All good, continue' on the first step before uploading an image will result in the tour immediately ending

### CODE REVIEW

**Comment by fditrapani** [comment]
I'm not seeing the final step for this one too.

File: client/layout/guided-tours/tours/checklist-user-avatar-tour.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+		<Step
+			name="click-done"
+			target="image-editor-button-done"
+			arrow="bottom-center"
+			placement="above"
+		>
+			<Continue target="image_editor_button_done" step="finish" click>
+				{ translate(
+					"Let's make sure it looks right before you press {{b}}Change My Photo{{/b}} to save your changes.",
```
Comment (a8ci18n) [inline client/layout/guided-tours/tours/checklist-user-avatar-tour.js]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Let's make sure it looks right before you press {{b}}Done{{/b}} to save your changes.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/layout/guided-tours/tours/checklist-user-avatar-tour.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+		<Step
+			name="click-done"
+			target="image-editor-button-done"
+			arrow="bottom-center"
+			placement="above"
+		>
+			<Continue target="image-editor-button-done" step="finish" click>
+				{ translate(
+					"Let's make sure it looks right before you press {{b}}Change My Photo{{/b}} to save your changes.",
```
Comment (a8ci18n) [inline client/layout/guided-tours/tours/checklist-user-avatar-tour.js]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Let's make sure it looks right before you press {{b}}Done{{/b}} to save your changes.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by dzver** [comment]
LGTM

**Comment by lsinger** [comment]
Not a blocker, but we should make the "dear framework, please never trigger me automatically" more readable. Documented here: #20914

**Comment by fditrapani** [comment]
I made a small update that prevents the guided tour from being displayed over the avatar. I also dropped the second step as it seemed a little awkward and unnecessary. 

I still can't get the last step to show on my end. Any ideas why?

**Pull request merged**